src/extension/providers/debug_config_provider.ts

```diff
@@ -672,6 +673,13 @@ export class DebugConfigProvider implements DebugConfigurationProvider {
 				if (renderer)
 					this.addArgsIfNotExist(args, "--web-renderer", renderer);
 			}
+
+			if (this.wsContext.config.forceFlutterWorkspace && conf.config.has("daemonPort")) {
+				const port = conf.config.get<number|undefined>("daemonPort");
```

Comment:
This should just be `config.daemonPort` and we should add a getter to `ResourceConfig` (this is the same as what you did originally, but in the `ResourceConfig` class instead of `Config`. One is global, and one supports resource-scoped (essentially workspace-folder/project scoped) settings.

Reply:
Oh is this why I was having trouble accessing `daemonPort` before? I'd only had it in the first config.

Reply:
I wouldn't have expected so. The only time these two should be different is in a multi-root workspace. Otherwise, it should always ready from `.vscode/settings.json` at the top level first, and then user settings if it's not defined there.

However, remote sessions do have some differences with config (in part to stop cross-platform paths being roamed, like trying to use Windows paths set in settings on a Linux remote). I don't know the full details, but it's possible that affected this (or that its presence and/or `"scope"` setting in our `package.json` affects it).